MapboxNavigation/InstructionPresenter.swift

```diff
@@ -166,7 +167,10 @@ class InstructionPresenter {
     }
 
     private func instructionHasDownloadedAllShields() -> Bool {
-        for component in instruction.textComponents {
+        let textComponents = instruction.components.compactMap { $0 as? VisualInstructionComponent }
+        guard !textComponents.isEmpty else { return false }
```

Comment:
This can be written more succinctly as:

```swift
guard let _ = instruction.components.contains(where: { $0 is VisualInstructionComponent }) else { return false }
```

Reply:
`guard instruction.components.contains(where: { $0 is VisualInstructionComponent }) else { return false }` 😏

Reply:
@1ec5 Indeed that is succinct. However, for the purpose of this function scope, we need the variable and also an early exit if its empty. Essentially, we need the contents of `textComponent` to iterate through at the proceeding [lines](https://github.com/mapbox/mapbox-navigation-ios/pull/1514/files/13f2d71321b7d5e320721e611c983e6411a2d904#diff-af48d70309ee195e34cc5e6cc6b27caaR173)